sdk/core/Azure.Core/src/Shared/HttpMessageSanitizer.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace Azure.Core
+{
+    internal class HttpMessageSanitizer
+    {
+        private readonly bool _logAllHeaders;
+        private readonly bool _logFullQueries;
+        private readonly string[] _allowedQueryParameters;
+        private readonly string _redactedPlaceholder;
+        private readonly HashSet<string> _allowedHeaders;
+        private const string LogAllValue = "*";
```

Comment:
Nit: we don't enforce it, but it's common to put consts before static readonlies.